tensorboard/plugins/hparams/list_session_groups.py

```diff
@@ -811,3 +814,45 @@ def _measurements(session_group, metric_name):
         if not metric_value:
             continue
         yield _Measurement(metric_value, session_index)
+
+
+def _build_data_provider_filters(col_params):
+    """Builds HyperparameterFilters from ColParams."""
+    filters = []
+    for col_param in col_params:
+        fltr = _build_data_provider_filter(col_param)
+        if fltr is None:
+            continue
+        filters.append(fltr)
+    return filters
+
+
+def _build_data_provider_filter(col_param):
+    """Builds HyperparameterFilter from ColParam.
+
+    Args:
+      col_param: ColParam that possibly contains filter information.
+
+    Returns:
+      None if col_param does not specify filter information.
+    """
+    if col_param.HasField("filter_regexp"):
+        filter_type = provider.HyperparameterFilterType.REGEX
+        fltr = col_param.filter_regexp
+    elif col_param.HasField("filter_interval"):
+        filter_type = provider.HyperparameterFilterType.INTERVAL
+        fltr = (
+            col_param.filter_interval.min_value,
```

Comment:
Does it make sense to add a check here for `min_value <= max_value`? Or should this be delegate to the server? (Also not sure if UI already has some sort of enforcement for this)

Reply:
Returning empty results in this case makes sense 👌